Segment-Optimizely-X/Classes/SEGOptimizelyXIntegration.m

```diff
@@ -0,0 +1,201 @@
+//
+//  SEGOptimizelyXIntegration.m
+//  Pods
+//
+//  Created by ladan nasserian on 8/17/17.
+//
+//
+
+#import "SEGOptimizelyXIntegration.h"
+#import <Analytics/SEGIntegration.h>
+#import <Analytics/SEGAnalyticsUtils.h>
+#import <Analytics/SEGAnalytics.h>
+
+
+@implementation SEGOptimizelyXIntegration
+
+#pragma mark - Initialization
+
+- (instancetype)initWithSettings:(NSDictionary *)settings andOptimizelyManager:(OPTLYManager *)manager withAnalytics:(SEGAnalytics *)analytics
+{
+    if (self = [super init]) {
+        self.settings = settings;
+        self.manager = manager;
+        self.analytics = analytics;
+        self.backgroundQueue = dispatch_queue_create("com.segment.integrations.optimizelyx.backgroundQueue", NULL);
+    }
+
+    if ([(NSNumber *)[self.settings objectForKey:@"listen"] boolValue]) {
+        self.observer = [[NSNotificationCenter defaultCenter] addObserverForName:OptimizelyDidActivateExperimentNotification
+                                                                          object:nil
+                                                                           queue:nil
+                                                                      usingBlock:^(NSNotification *_Nonnull note) {
+                                                                          [self experimentDidGetViewed:note];
+                                                                      }];
+    }
+
+    return self;
+}
+
+
+- (void)identify:(SEGIdentifyPayload *)payload
+{
+    if (payload.userId) {
+        self.userId = payload.userId;
+        SEGLog(@"SEGOptimizelyX assigning userId %@", self.userId);
+    }
+
+    if (payload.traits) {
+        self.userTraits = payload.traits;
+        SEGLog(@"SEGOptimizelyX assigning attributes %@", self.userTraits);
+    }
+}
+
+
+- (void)track:(SEGTrackPayload *)payload
+{
+    if ([self.manager getOptimizely] == nil) {
+        [self enqueueAction:payload];
+    }
+    self.client = [self.manager getOptimizely];
```

Comment:
self.client is only used in this method. Instead of making it a global variable, you can make it a local variable.

```
OPTLYClient *client = [self.manager getOptimizely];
```

Reply:
Hmm it is used throughout the implementation - self.client is the instance of Optimizely I call Optimizely calls on

Reply:
I don't think it is. https://cloudup.com/cBWBaFAawlS I see seven instances fro my search, 2 of which were in this github comment, and the other 5 were in this method.

If you make the change I suggest, and remove https://github.com/segment-integrations/analytics-ios-integration-optimizely-x/pull/1/files/2671479edfffa1deda6d23b4d83f8253d98409e6#diff-534dbac01cddf058612a49f729d8a3dcR19, does anything break?

Reply:
Ah gotcha - yeah that works